### PR TITLE
expose Encrypt-Format-All and Extension publisher/name options

### PR DESCRIPTION
**Pull request by vermashi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 1]
```diff
@@ -47,7 +47,10 @@ def encrypt_vm(resource_group_name, vm_name,  # pylint: disable=too-many-locals,
                key_encryption_keyvault=None,
                key_encryption_key=None,
                key_encryption_algorithm='RSA-OAEP',
-               volume_type=None):
+               volume_type=None,
+               encrypt_format_all=False,
+               extension_name=None,
+               extension_publisher=None):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: any reason we want to expose this?

Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: The latter. This is to allow the dev team and certain partners to check out test versions of the ADE extension.

Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: I suggest not to confuse end users, maybe consider other means such as env variables?

Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: I remember discussing this with the team as a whole- and we had ended up agreeing on the parameter input approach. If you have some other commands also using environment variables to target test-versions as examples to follow, I would be willing to take a look.

**Comment by vermashi** [comment]
From the build error, it seems like I need to bump the version for the module. Is my understanding correct? If so, how do I go about doing that?

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py (hunk 1 above)
Comment (yugangw-msft) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Seriously, I don’t think we should do this.  The implementation and testing related details are not meant to be exposed at command arguments to end users. 
I suggest it is the time you can use https://github.com/Azure/azure-cli-extensions. Once it is stabilized after partners' testing you can move the code back to azure-cli

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 2]
```diff
@@ -180,7 +193,10 @@ def encrypt_vm(resource_group_name, vm_name,  # pylint: disable=too-many-locals,
                        "the encryption will finish shortly")
 
 
-def decrypt_vm(resource_group_name, vm_name, volume_type=None, force=False):
+def decrypt_vm(resource_group_name,
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: NIT: are you using any python editor to format this line? We have updated pylint configuration to allow bigger line width, so you should back out

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 3]
```diff
@@ -39,6 +40,15 @@
     }
 }
 
+def _get_extension_info(os_type):
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: I suggest we remove this method, rather just apply on the dictionary above such as :

```python
vm_extension_info = {
     'Linux': {
         'publisher': os.environ.get('ADE_TEST_EXTENSION_PUBLISHER') or 'Microsoft.Azure.Security',
         'name': os.environ.get('ADE_TEST_EXTENSION_NAME') or 'AzureDiskEncryptionForLinux',
         'version': '0.1'
     },
     'Windows': {
         'publisher': os.environ.get('ADE_TEST_EXTENSION_PUBLISHER') or 'Microsoft.Azure.Security',
         'name': os.environ.get('ADE_TEST_EXTENSION_NAME') or 'AzureDiskEncryption',
         'version': '1.1'
     }
 }
 ```

**Comment by yugangw-msft** [comment]
@vermashi, please rebase and resolve conflicts for me to merge

**Comment by vermashi** [comment]
@yugangw-msft Done. :-)

**Comment by vermashi** [comment]
Actually, hold on. Made a small mistake while rebasing.

File: src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 import uuid
+import os
 from msrestazure.tools import parse_resource_id
```
Comment (vermashi) [inline src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: @yugangw-msft 
My local build seems to be breaking on this import. Any idea what's up?

Comment (vermashi) [reply src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/disk_encryption.py]: Nevermind, it was a local build issue

**Comment by yugangw-msft** [comment]
Please address the pylint error:
`./src/command_modules/azure-cli-vm/build/lib/azure/cli/command_modules/vm/disk_encryption.py:44:1: E302 expected 2 blank lines, found 1`

**Pull request merged**